patches/@react-native-async-storage+async-storage+1.23.1.patch

```diff
@@ -0,0 +1,314 @@
+diff --git a/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.h b/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.h
+index 38e5178..9a87331 100644
+--- a/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.h
++++ b/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.h
+@@ -42,12 +42,12 @@
+ 
+ @property (nonatomic, readonly, getter=isValid) BOOL valid;
+ 
++// NOTE(nikki): Added to allow scoped per Expo app
++- (instancetype)initWithStorageDirectory:(NSString *)storageDirectory;
++
+ // Clear the RNCAsyncStorage data from native code
+ - (void)clearAllData;
+ 
+-// For clearing data when the bridge may not exist, e.g. when logging out.
+-+ (void)clearAllData;
+-
+ // Grab data from the cache. ResponseBlock result array will have an error at position 0, and an
+ // array of arrays at position 1.
+ - (void)multiGet:(NSArray<NSString *> *)keys callback:(RCTResponseSenderBlock)callback;
+diff --git a/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.mm b/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.mm
+index 6b81187..d61ceff 100644
+--- a/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.mm
++++ b/node_modules/@react-native-async-storage/async-storage/ios/RNCAsyncStorage.mm
+@@ -14,7 +14,9 @@
+ #import <React/RCTLog.h>
+ #import <React/RCTUtils.h>
+ 
+-static NSString *const RCTStorageDirectory = @"RCTAsyncLocalStorage_V1";
++// NOTE(kudo): Use Expo storage directory for backward compatibility
++//static NSString *const RCTStorageDirectory = @"RCTAsyncLocalStorage_V1";
++static NSString *const RCTStorageDirectory = @"RCTAsyncLocalStorage";
```

Comment:
please remember to remove this before merge. this patch could also affect to bare-expo